language-examples/remedial-js/9.0-ES5-vs-ES6/README.md

```diff
@@ -0,0 +1,943 @@
+
+ES6:
+
+```js
+class Hello {
+  constructor(name) {
+    this.name = name;
+  }
+
+  hello() {
+    return 'Hello ' + this.name + '!';
+  }
+
+  static sayHelloAll() {
+    return 'Hello everyone!';
+  }
+}
+
+class HelloWorld extends Hello {
+  constructor() {
+    super('World');
+  }
+
+  echo() {
+    alert(super.hello());
+  }
+}
+
+var hw = new HelloWorld();
+hw.echo();
+
+alert(Hello.sayHelloAll());
+```
+
+ES5 (approximate):
+
+```js
+function Hello(name) {
+  this.name = name;
+}
+
+Hello.prototype.hello = function hello() {
+  return 'Hello ' + this.name + '!';
+};
+
+Hello.sayHelloAll = function () {
+  return 'Hello everyone!';
+};
+
+function HelloWorld() {
+  Hello.call(this, 'World');
+}
+
+HelloWorld.prototype = Object.create(Hello.prototype);
+HelloWorld.prototype.constructor = HelloWorld;
+HelloWorld.sayHelloAll = Hello.sayHelloAll;
+
+HelloWorld.prototype.echo = function echo() {
+  alert(Hello.prototype.hello.call(this));
+};
+
+var hw = new HelloWorld();
+hw.echo();
+
+alert(Hello.sayHelloAll());
+```
+
+A more faithful (albeit, slightly verbose) interpretation can be found in this [Babel](https://goo.gl/ZvEQDq) output.
+
+## Modules
+
+Modules are mostly implemented, with some parts of the Loader API still to be corrected. Modules try to solve many issues in dependencies and deployment, allowing users to create modules with explicit exports, import specific exported names from those modules, and keep these names separate.
+
+*Assumes an environment using CommonJS*
+
+
+app.js - ES6
+
+```js
+import math from 'lib/math';
+console.log('2π = ' + math.sum(math.pi, math.pi));
+```
+
+app.js - ES5
+
+```js
+var math = require('lib/math');
+console.log('2π = ' + math.sum(math.pi, math.pi));
+```
+
+lib/math.js - ES6
+
+```js
+export function sum(x, y) {
+  return x + y;
+}
+export var pi = 3.141593;
+```
+
+lib/math.js - ES5
+
+```js
+exports.sum = sum;
+function sum(x, y) {
+  return x + y;
+}
+var pi = exports.pi = 3.141593;
+```
+
+lib/mathplusplus.js - ES6
+
+```js
+export * from 'lib/math';
+export var e = 2.71828182846;
+export default function(x) {
+  return Math.exp(x);
+}
+```
+
+lib/mathplusplus.js - ES5
+
+```js
+var Math = require('lib/math');
+
+var _extends = function (target) {
+  for (var i = 1; i < arguments.length; i++) {
+    var source = arguments[i];
+    for (var key in source) {
+      target[key] = source[key];
+    }
+  }
+
+  return target;
+};
+
+var e = exports.e = 2.71828182846;
+exports['default'] = function (x) {
+  return Math.exp(x);
+};
+
+module.exports = _extends(exports['default'], exports);
+```
+
+## Numeric Literals
+
+ES6:
+
+```js
+var binary = [
+  0b0,
+  0b1,
+  0b11
+];
+console.assert(binary === [0, 1, 3]);
+
+var octal = [
+  0o0,
+  0o1,
+  0o10,
+  0o77
+];
+console.assert(octal === [0, 1, 8, 63]);
+```
+
+ES5:
+
+```js
+'use strict';
+
+var binary = [0, 1, 3];
+console.assert(binary === [0, 1, 3]);
+
+var octal = [0, 1, 8, 63];
+console.assert(octal === [0, 1, 8, 63]);
+```
```

Comment:
I think this is fine, I like the using parseInt suggestion as it makes the ES5 comparison more relevant.  I updated to use parseInt (for the ES5 comparison)